### PR TITLE
Fix issue #4482

### PR DESCRIPTION
**Pull request by tomz**
If location is on s3, skip the path existence and directory checking because the directory may not exist yet and will be created.

### CODE REVIEW

File: presto-hive/src/main/java/com/facebook/presto/hive/HiveWriteUtils.java [hunk 1]
```diff
@@ -348,11 +348,13 @@ public static Path getTableDefaultLocation(String user, HiveMetastore metastore,
         }
 
         Path databasePath = new Path(location);
-        if (!pathExists(user, hdfsEnvironment, databasePath)) {
-            throw new PrestoException(HIVE_DATABASE_LOCATION_ERROR, format("Database '%s' location does not exist: %s", schemaName, databasePath));
-        }
-        if (!isDirectory(user, hdfsEnvironment, databasePath)) {
-            throw new PrestoException(HIVE_DATABASE_LOCATION_ERROR, format("Database '%s' location is not a directory: %s", schemaName, databasePath));
+        if (!location.startsWith("s3")) {
```
Comment (electrum) [inline presto-hive/src/main/java/com/facebook/presto/hive/HiveWriteUtils.java]: Comparing the URI is brittle and will break if a new URI scheme is registered for S3. There is a method `HiveLocationService.shouldUseTemporaryDirectory()` that already does the check correctly.

Let's create a new helper method in `HiveUtil` named `isS3FileSystem` using the code from `shouldUseTemporaryDirectory`, then have `shouldUseTemporaryDirectory` call the new method.

Then we can use the new method here as well.

**Comment by electrum** [comment]
@nezihyigitbasi @zhenxiao does this approach seem fine to you?

**Comment by zhenxiao** [comment]
I like @electrum 's comment, should not compare string, adding in HiveUtils is better.

**Comment by tomz** [comment]
Ok thanks for recommendations, I will make the change

**Comment by tomz** [comment]
@electrum code refactored based on comments

**Comment by tomz** [comment]
@electrum not sure what the build didn't complete, looked like some kind of timeout not related to the change

File: presto-hive/src/main/java/com/facebook/presto/hive/HiveLocationService.java [hunk 2]
```diff
@@ -21,13 +21,13 @@
 
 import javax.inject.Inject;
 
-import java.io.IOException;
 import java.util.Optional;
 
-import static com.facebook.presto.hive.HiveErrorCode.HIVE_FILESYSTEM_ERROR;
+//import static com.facebook.presto.hive.HiveErrorCode.HIVE_FILESYSTEM_ERROR;
```
Comment (nezihyigitbasi) [inline presto-hive/src/main/java/com/facebook/presto/hive/HiveLocationService.java]: you can remove this comment

**Comment by nezihyigitbasi** [comment]
LGTM. Can you please add a unit test?

File: presto-hive/src/test/java/com/facebook/presto/hive/TestHiveWriteUtils.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+import static org.testng.Assert.assertEquals;
+
+public class TestHiveWriteUtils
+{
+    @Test
+    public void testIsS3FileSystem()
+    {
+        HdfsEnvironment hdfsEnvironment = createTestHdfsEnvironment(new HiveClientConfig());
+        assertEquals(isS3FileSystem("user", hdfsEnvironment, new Path("s3://test-bucket/test-folder")), true);
```
Comment (electrum) [inline presto-hive/src/test/java/com/facebook/presto/hive/TestHiveWriteUtils.java]: When comparing to true/false, use `assertTrue` or `assertFalse`. I'll fix this up and merge.

**Comment by electrum** [comment]
Merged, thanks!

**Pull request closed without merging**